wrapper/sessions.go

```diff
@@ -295,6 +311,7 @@ func (s *Session) Disconnect(code int) error {
 		}
 	}
 
+	s.cancel()
```

Comment:
newline before return
```suggestion
	s.cancel()
	
```